BlueprintUI/Sources/Element/ElementContent.swift

```diff
@@ -287,8 +301,27 @@ extension ElementContent {
             return result
         }
 
-        private func layoutItems(in environment: Environment) -> [(LayoutType.Traits, Measurable)] {
-            return children.map { ($0.traits, $0.content.measurable(in: environment)) }
+        private func layoutItems(
+            in environment: Environment,
+            cache: CacheTree
+        ) -> [(LayoutType.Traits, Measurable)] {
+            return zip(children.indices, children).map { index, child in
```

Comment:
Nit: When I was profiling, `zip` is for some reason slower than just doing `foo.enumerated()`, I'd recommend swapping out to that throughout the PR.